x-pack/plugins/siem/public/containers/case/use_get_case_user_actions.tsx

```diff
@@ -95,6 +127,18 @@ export const getPushedInfo = (
               firstPushIndex: i,
               lastPushIndex: i,
               hasDataToPush: hasDataToPushForConnector(externalService.connectorId),
+              commentsToUpdate: commentsAndIndex.reduce<string[]>(
```

Comment:
I think it's better to move this procedure outside of the `caseUserActions.reduce` and pass again the `caseServices` to add `commentsToUpdate`. I am not sure if we need it inside or not. Do we really need to do it for every `i` ?

Reply:
yes because we are checking if that comment happened after the last push, `i`. `currentComment.commentIndex > i`

Reply:
Yes, but we do not need to do it for every `i`. We only need to know the lastPushedIndex. All comments after that are included. 

We can do this after the first reduce

```
caseServices = Object.keys(caseServices).reduce<CaseServices>((acc, key) => {
    return {
      ...acc,
      [key]: {
        ...caseServices[key],
        commentsToUpdate: buildCommentsToUpdate(commentsAndIndex, caseServices[key].lastPushIndex),
      },
    };
  }, {});
```

Reply:
done!